addons/udes_stock/models/stock_picking.py

```diff
@@ -0,0 +1,175 @@
+
+        new_moves = Move.browse()
+
+        for current_move in mls.mapped('move_id'):
+            current_mls = mls.filtered(lambda x: x.move_id == current_move)
+            new_moves |= current_move.split_out_move_lines(current_mls)
+
+        # Create picking for completed move lines
+        bk_picking = self.copy(
+            {'name': '/', 'move_lines': [], 'move_line_ids': [], 'backorder_id': self.id,}
+        )
+        new_moves.write({'picking_id': bk_picking.id})
+        new_moves.mapped('move_line_ids').write({'picking_id': bk_picking.id})
+
+        return bk_picking
+
+    def _requires_backorder(self, mls):
+        """ Checks if a backorder is required by checking if all move.lines
+            within a picking are present in mls
+            TODO: Consolidate with _check_backorder in Odoo core - figure out cases _check_backorder misses
+        """
+        mls_moves = mls.mapped('move_id')
+        for move in self.move_lines:
+            if (
+                move not in mls_moves
+                or not move.move_line_ids == mls.filtered(lambda x: x.move_id == move)
+                or move.move_orig_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+            ):
+                return True
+        return False
+
+    def create_picking(
+        self,
+        picking_type,
+        products_info,
+        location_id=None,
+        location_dest_id=None,
+        confirm=False,
+        assign=False,
+        create_batch=False,
+        **kwargs,
+    ):
+        """ Create and return a picking for the given picking_type
+            Function based on test function in Odoo11, with some notions
+            retained from craet_picking in udes_stock
+            :args:
+                - picking_type
+                - products_info
+            :kwargs:
+                - location_id
+                - location_dest
+                - confirm
+                - asign
+                - create_batch
+        """
+        Picking = self.env['stock.picking']
+        # if location_id not specified, revert to default for that picking_type
+        if location_id is None:
+            location_id = picking_type.default_location_src_id
+        if location_dest_id is None:
+            location_dest_id = picking_type.default_location_dest_id
+
+        # Create stock.picking
+        vals = {
+            'picking_type_id': picking_type.id,
+            'location_id': location_id.id,
+            'location_dest_id': location_dest_id.id,
+        }
+
+        vals.update(kwargs)
+        picking = Picking.create(vals)
+
+        # Create stock.moves
+        for product_info in products_info:
+            product_info.update(picking=picking)
+            self._create_move(**product_info)
+
+        if confirm:
+            picking.action_confirm()
+
+        if assign:
+            picking.action_assign()
+
+        if create_batch:
+            # stock.picking.batch does not exist- so batch create does not yet
```

Comment:
This could be .create() instead of .create_batch() and change it once create_batch() exists.